File: interpreter-for-cpp/src/Runtime/Sentence/SentenceLoopControl.cpp
#include "SentenceLoopControl.h"

using namespace peak::interpreter;

ExecuteResult SentenceLoopControlContinue::Execute(std::shared_ptr<Space> space) {
	return ExecuteResult::Continue;
}
ExecuteResult SentenceLoopControlBreak::Execute(std::shared_ptr<Space> space) {
	return ExecuteResult::Break;
}

File: interpreter-for-cs/src/Runtime/Sentence/SentenceLoopControl.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {
	class SentenceLoopControlContinue : Sentence {
		public override ExecuteResult Execute(Space space) {
			return ExecuteResult.Continue;
		}
	}
	class SentenceLoopControlBreak : Sentence {
		public override ExecuteResult Execute(Space space) {
			return ExecuteResult.Break;
		}
	}
} // namespace peak.interpreter